### DISCUSSION TITLE
compatible versions of PyTorch and torchkbnufft.

### DISCUSSION THREAD

**Discussion by RaziehEmami1** [mmuckley/torchkbnufft#59] [General]
Hi everyone, 

I am trying to make a image using torch version 1.10.0 in CUDA and I am hoping to use torchkbnufft as a part of the analysis. I was wondering what is the compatible version of this with torchkbnufft. Otherwise the code crashes and is unable to call fft. I am currently trying torchkbnnufft== 1.3.0 in my remote builder. 

Thanks a lot in advance. 
RE.

**Comment by mmuckley** [comment]
I think it should work with PyTorch 1.10. We test using PyTorch 1.10 in our continuous integration. Could you try installing from master and seeing if you still have the fft error?

`pip install --upgrade git+https://github.com/mmuckley/torchkbnufft`

**Reply by RaziehEmami1** [reply]
Hi 

Unfortunately, it did not work. I used 

pip install torch==1.10.1+cu111 torchvision==0.11.2+cu111 torchaudio==0.10.1 -f https://download.pytorch.org/whl/torch_stable.html
  pip install torchkbnufft==1.3.0 

and when I try to import torchkbnufft it says that it is not callable. I also tried
 pip install torchkbnufft==0.2.0  
in my container and this time it is able to import them but when we use nnufft it fails. 

May I ask what you recommend, please?

**Reply by mmuckley** [reply]
Hello @RaziehEmami1 I recommended to install from main - but your command indicates you installed from PyPI, could you try running the install command I posted?

**Reply by RaziehEmami1** [reply]
Hi @mmuckley 
many thanks for your kind guide. I will absolutely try and let you know what I get. Just to be sure, do not we need to specify the exact version of torchkbufft here? 

Thanks,
Razi

**Reply by mmuckley** [reply]
No - that command will just pull whatever's in the main branch of GitHub. That way I can make sure that the code which is being tested in continuous integration is also not working for you.

**Reply by RaziehEmami1** [reply]
Fantastic. 
Many thanks for your kind helps. I am trying now and will let you know what I get. 

Cheers,
Razi

**Reply by RaziehEmami1** [reply]
Hi
so I tried with the version you said and here is the error that I received

  from torchkbnufft import KbNufft, AdjKbNufft
ImportError: cannot import name 'AdjKbNufft' from 'torchkbnufft' (/opt/conda/lib/python3.9/site-packages/torchkbnufft/__init__.py)


I wonder does is have anythings to do with the python version or is related to the incompatibilities between the version of the pytorch and torchkbnufft? 

Let me add that I am using A100 in the cluster. I wonder could that be that this is incompatible with the torch and torchkbnufft that we wish to use? 

thanks.

**Reply by mmuckley** [reply]
Hello @RaziehEmami1, there is no `AdjKbNufft` in the package. That class is called `KbNufftAdjoint` (possibly renamed in a past version).

**Comment by RaziehEmami1** [comment]
Hi,

Thanks a lot for your kind helps. Yes, it is working now. 

The new complain goes to the other lines of the code, as they seem to be renamed in the newer version

from torchkbnufft.mri.dcomp_calc import calculate_radial_dcomp_pytorch
from torchkbnufft.math import absolute

from torch.utils.data import DataLoader, Dataset

I was wondering may I ask your opinion about them, please? 


Here is the error:

from torchkbnufft.mri.dcomp_calc import calculate_radial_dcomp_pytorch
ModuleNotFoundError: No module named 'torchkbnufft.mri'


Thanks a lot,
Razi

**Reply by mmuckley** [reply]
Please consult the documentation on [readthedocs](https://torchkbnufft.readthedocs.io/en/stable/), which should be up-to-date.
